### PR TITLE
Support DMX over FTDI USB devices

### PR DESCRIPTION
**Pull request by xian**
- Include ftd2xx and JNI driver for Mac.
- Abstract some stuff up from `Shenzarpy` into generic `MovingHead`.

### CODE REVIEW

File: src/jvmMain/kotlin/baaahs/dmx/DmxDevice.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        @Throws(FTD2xxException::class)
+        private fun transmit(bytes: ByteArray) {
+            val port = device.port
+            port.setBreakOn(true)
+            port.setBreakOn(false)
+            device.write(bytes)
+
+            sleep(DMX_TRANSMIT_DELAY)
```
Comment (xian) [inline src/jvmMain/kotlin/baaahs/dmx/DmxDevice.kt]: This should probably happen on a timer every _n_ ms instead of with an absolute delay, right?

Comment (joshdurbin) [reply src/jvmMain/kotlin/baaahs/dmx/DmxDevice.kt]: ...via a ticker routine?

Comment (xian) [reply src/jvmMain/kotlin/baaahs/dmx/DmxDevice.kt]: Yeah... per  [this](https://docs.lumenrad.io/crmxchip/dmx-interface/#dmx-frame-rate-and-size) it sounds like we're within tolerances.

**Comment by xian** [comment]
This is missing the JNI driver for linux but I think otherwise good to go.

**Pull request merged**